### PR TITLE
Proposed changes to fix halo_mass bugs

### PR DESCRIPTION
**Pull request by evevkovacs**
@yymao  Here are the changes to fix the halo_mass units issue. I sent you an email about my attempt to fix the units reported by get_quantity_info. There may be a better way. Please let me know.

### CODE REVIEW

File: GCRCatalogs/cosmodc2.py [hunk 1]
```diff
@@ -451,6 +451,13 @@ def _generate_quantity_modifiers(self):
 
         return quantity_modifiers
 
+    def get_quantity_info(self,q):
+        info_dict = self._get_quantity_info_dict(q)
+        if q == 'hostHaloMass':
+            if 'units' in info_dict:
+                info_dict['units'] = 'Msun/h'
+        
+        return info_dict
```
Comment (yymao) [inline GCRCatalogs/cosmodc2.py]: We should update the upstream method instead. 

```python
    def _collect_native_quantities(self, fh, collect_info_dict=False):
        orig_output = super()._collect_native_quantities(fh, collect_info_dict)
        # a hot fix of the unit of native halo mass (hostHaloMass), which should be Msun/h
        if collect_info_dict:
            orig_output[1]['hostHaloMass']['unit'] = 'Msun/h'
        return output
```

Comment (evevkovacs) [reply GCRCatalogs/cosmodc2.py]: OK, this is now implemented and tested. This PR is ready to merge. baseDC2 does not have units so no modifications are necessary. protoDC2 remains to be checked.

**Comment by evevkovacs** [comment]
@yymao  Oh, yes, I suppose we do since these are schema quantities. There are no unit designations to fix however. I'll make that change and re-commit. I'll need to fix this for the snapshot baseDC2 as well. Thanks for catching this.

**Comment by evevkovacs** [comment]
@yymao  I fixed the quantity modifier for alphaq.py (protoDC2 reader) and added the code you suggested for modifying the units for cosmoDC2. That is, I defined _collect_native_quantities(self, fh, collect_info_dict=False) etc. But when I tested on protoDC2, it didn't work. I still got units of Msun for hostHaloMass. Does this function exist for the AlphaQGalaxyCatalog class? Should I use something else? Thanks

**Comment by yymao** [comment]
@evevkovacs protoDC2 (`alphaq.py`) uses a different method to collect the quantity info. We will need to modify the `_get_native_quantity_info_dict` method. I'll be happy to make another PR for it, but can you confirm that protoDC2 also needs the fix 
I have thought that protoDC2 was schema conforming, but I could be wrong. 

In any case, this PR is indeed ready to be merged.

**Comment by evevkovacs** [comment]
OK will merge now and address other issues next.

**Pull request merged**